File: Python/01_Strings_Arrays/IsPalindromePerm.py
def is_palindrome_perm(string):
    #print(string)
    counts = [False]*128
    for i in range(0, len(string)):
        index = ord(string[i])
        counts[index] = not counts[index]

    oddFound = False
    for i in range(0, 128):
        if counts[i] and oddFound:
            return False

        if counts[i]:
            oddFound = True

    return True


print(is_palindrome_perm("abba")) #true
print(is_palindrome_perm("abcba")) # true
print(is_palindrome_perm("abbac"))#true
print(is_palindrome_perm("ab")) #false
print(is_palindrome_perm("a")) #true
print(is_palindrome_perm("abcde")) #false
print(is_palindrome_perm("abcda")) #false

File: C++/Practice_Pad/source/CH1/is_palindrome_perm.cpp
#include "pch.h"
#include "is_palindrome_perm.h"

bool is_palindrome_perm(string s) {
    cout << "String: " << s;

    bool counts[128] = { false };
    for (int i = 0; i < s.length(); i++) {
        int index = s.at(i);
        counts[index] = !counts[index];
    }
    bool oddFound = false;
    for (int i = 0; i < 128; i++) {
        if (counts[i] && oddFound) {
            cout << " False " << endl;
            return false;
        }
        if (counts[i]) {
            oddFound = true;
        }
    }
    cout << " True" << endl;
    return true;
}